w3af/plugins/crawl/soap.py

```diff
@@ -0,0 +1,168 @@
+import contextlib
+import sys
+from cStringIO import StringIO
+
+import zeep
+from zeep.exceptions import XMLSyntaxError
+
+import w3af.core.data.kb.knowledge_base as kb
+from w3af.core.controllers import output_manager
+from w3af.core.controllers.plugins.crawl_plugin import CrawlPlugin
+from w3af.core.data.kb.info import Info
+from w3af.core.data.options.opt_factory import opt_factory
+from w3af.core.data.options import option_types
+from w3af.core.data.parsers.doc.url import URL
+from w3af.core.data.request.fuzzable_request import FuzzableRequest
+
+
+@contextlib.contextmanager
+def redirect_stdout(new_stdout):
+    old_stdout = sys.stdout
+    try:
+        sys.stdout = new_stdout
+        yield
+    finally:
+        sys.stdout = old_stdout
+
+
+class soap(CrawlPlugin):
+    def __init__(self):
+        super(soap, self).__init__()
+        self._is_first_run = True
+        self.service_url = ''
+        self.verbosity_level = 0
+
+    def crawl(self, fuzzable_request, debugging_id):
+        url_to_crawl = fuzzable_request.get_url().url_string
+        if self._is_first_run:
+            self._is_first_run = False
+            url_to_crawl = self.service_url
```

Comment:
On the first call you lose the `fuzzable_request` URL. Instead of doing that, maybe you can perform the analysis twice, once on the `self.service_url` and once on `fuzzable_request`.

It is very unlikely, but maybe the `fuzzable_request` contains valuable data?

Reply:
Not longer an issue, SOAP plugin is refactored to `wsd_parser`